centreon/www/include/options/media/images/DB-Func.php

```diff
@@ -174,22 +174,37 @@ function deleteImg($imageId)
     global $pearDB;
 
     $mediadir = "./img/media/";
-
-    $dbResult = $pearDB->query(
-        "SELECT dir_alias, img_path "
-        . "FROM view_img, view_img_dir, view_img_dir_relation "
-        . "WHERE img_id = $imageId AND img_id = img_img_id "
-        . "AND dir_dir_parent_id = dir_id"
-    );
-    while ($imagePath = $dbResult->fetch()) {
-        $fullpath = $mediadir . basename($imagePath["dir_alias"]) . "/" . basename($imagePath["img_path"]);
-        if (is_file($fullpath)) {
-            unlink($fullpath);
+    try {
+        $pearDB->beginTransaction();
+        $query = "SELECT dir_alias, img_path
+          FROM view_img, view_img_dir, view_img_dir_relation
+          WHERE img_id = :imageId
+          AND img_id = img_img_id
+          AND dir_dir_parent_id = dir_id";
+
+        $dbResult = $pearDB->prepare($query);
+        $dbResult->bindParam(':imageId', $imageId, PDO::PARAM_INT);
+        $dbResult->execute();
+        foreach ($dbResult->fetchAll(PDO::FETCH_ASSOC) as $imagePath) {
+            $fullpath = $mediadir . basename($imagePath["dir_alias"]) . "/" . basename($imagePath["img_path"]);
+            if (is_file($fullpath)) {
+                unlink($fullpath);
+            }
+            $deleteQuery  = "DELETE FROM view_img WHERE img_id = :imageId";
+            $deleteStatement = $pearDB->prepare($deleteQuery);
+            $deleteStatement->bindParam(':imageId', $imageId, PDO::PARAM_INT);
+            $deleteStatement->execute();
+
+            $deleteRelationQuery = "DELETE FROM view_img_dir_relation WHERE img_img_id = :imageId";
+            $deleteRelationStatement = $pearDB->prepare($deleteRelationQuery);
+            $deleteRelationStatement->bindParam(':imageId', $imageId, PDO::PARAM_INT);
+            $deleteRelationStatement->execute();
         }
-        $pearDB->query("DELETE FROM view_img WHERE img_id = $imageId");
-        $pearDB->query("DELETE FROM view_img_dir_relation WHERE img_img_id = $imageId");
+        $pearDB->commit();
```

Comment:
I doubt it makes sense to loop on these two queries while the var passed to the placeholder ($imageId) is defined above the foreach and thus never changes.

You just loop on the exact same query with the exact same parameter.
Those two queries should probably move outside the foreach part.

(I know it was already the previous behaviour but let's still fix it)